metapool/metapool.py

```diff
@@ -2075,3 +2077,43 @@ def validate_plate_df(
             "The following sample names are duplicated %s"
             % ", ".join(sorted(duplicated_samples))
         )
+
+
+def generate_override_cycles_value(runinfo_xml_path, adapter_length):
+    if adapter_length < 0:
+        raise ValueError("Adapter-length cannot be less than zero")
+
+    tree = ET.parse(runinfo_xml_path)
+    reads = tree.getroot().findall('.Run/Reads/Read')
+    results = []
+    for read in reads:
+        result = {'number': int(read.get('Number')),
+                  'num_cycles': int(read.get('NumCycles'))}
+        result['is_indexed'] = True if read.get('IsIndexedRead') == 'Y' \
+            else False
+        results.append(result)
+
+    if len(results) == 0:
+        raise ValueError("Reads information could not be found in "
+                         f"'{runinfo_xml_path}'")
+
+    codes = []
+    # results should be in sorted order in XML file but this ensures that
+    # we process the elements in the correct order.
+    for read in sorted(results, key=itemgetter('number')):
+        if read['is_indexed'] is True:
+            code = f"I{adapter_length}"
+            truncated = read['num_cycles'] - adapter_length
+            if truncated < 0:
+                raise ValueError(f"num_cycles '{read['num_cycles']}' appears "
+                                 "to be less than adapter-length "
+                                 f"'{adapter_length}'")
+            elif truncated > 0:
+                code += f"N{truncated}"
+            # if truncated == 0 then a truncate code does not to be appended.
```

Comment:
can you update this comment?

Reply:
the comment is actually correct as-is. If the length of an adapter sequence is 8, and the value of NumCycles is also 8, then there is no need to append a truncate code. E.g. I8N2 if NumCycles is 10, and I8 if NumCycles is 8 (assuming the adapter length is also 8).